frontend/src/components/search/SearchFilterAccordion/SearchFilterAccordion.tsx

```diff
@@ -41,7 +47,12 @@ export function SearchFilterAccordion({
     toggleSelectAll,
     incrementTotal,
     decrementTotal,
-  } = useSearchFilter(initialFilterOptions);
+  } = useSearchFilter(
+    initialFilterOptions,
+    initialQueryParams,
+    queryParamKey,
+    formRef,
+  );
```

Comment:
More logic was added to `useSearchFilter` since the `SearchFilterAccordion` is getting sizeable